source/logging/logging.rst

```diff
@@ -0,0 +1,380 @@
+  variables.
+
+  If ``MONGODB_LOG_ALL`` is specified in addition to one or more component variables, the
+  component variable(s) MUST take precedence.
+
+  The default is to not log anything. 
+
+  If a variable is set to an invalid value, it MUST be treated as if it were not specified at all.
+
+Configurable Log Destination
+----------------------------
+Drivers MUST support configuring where log messages should be output, including the options: stdout,
+stderr, output file (path MUST be configurable).
+
+    **Fallback implementation method**: If the environment variable ``MONGODB_LOG_PATH`` is provided:
+     -  If the value is "stdout" (case-insensitive), log to stdout.
+     - If the value is "stderr" (case-insensitive), log to stderr.
+     - Else, log to a file at the specified path. If the file already exists, it MUST be appended to.
+
+    If the variable is not provided, the driver MUST log to stderr.
+
+Configurable Max Document Length
+--------------------------------
+Drivers MUST support configuring the maximum logged length for extended JSON documents in log messages.
+The unit here is flexible and can be bytes, Unicode code points, or graphemes, depending on what a
+driver is able to accomplish with its language's string APIs. The default max length is 1000 of
+whichever unit is selected.
+If the chosen unit is bytes, the driver MUST ensure that it gracefully handles cases where the
+truncation length falls mid code point, by either rounding the length up or down to the closest
+code point boundary or using the Unicode replacement character, to avoid producing invalid Unicode
+data.
+
+    **Fallback Implementation method**: Environment variable ``MONGOB_LOG_MAX_DOCUMENT_LENGTH``.
+    When unspecified, any extended JSON representation of a document which is longer than the
+    default max length MUST be truncated to that length. 
+    When set to an integer value, any extended JSON document longer than that value MUST be
+    truncated to that length.
+    If the variable is set to an invalid value, it MUST be treated as if it were not specified at
+    all.
+
+Components
+----------
+Drivers MUST support configuring minimum log severity levels on a per-component level. The below
+components currently exist and correspond to the listed specifications. This list is expected to
+grow over time as logging is added to more specifications.
+
+Drivers SHOULD specify the component names in whatever the idiomatic way is for their language.
+For example, the Java command component could be named ``org.mongodb.driver.command``.
+
+Drivers MAY define additional language-specific components in addition to these for any
+driver-specific messages they produce.
+
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 1
+
+   * - Component Name
+     - Specification(s)
+     - Environment Variable
+
+   * - command
+     - `Command Monitoring <../command-monitoring/command-monitoring.rst>`__
+     - ``MONGODB_LOG_COMMAND``
+
+   * - sdam
```

Comment:
Let's avoid this abbreviation.  Very few people reading the log files will know what it refers to.  How about `topology` instead?